src/StoreKeeper/WooCommerce/B2C/PaymentGateway/PaymentGateway.php

```diff
@@ -284,97 +282,44 @@ public function onReturn()
-                }
-            }
-        } catch (GeneralException $generalException) {
-            if ('Only invoiced orders can be refunded' === $generalException->getMessage()) {
-                self::createRefundAsPayment($orderId, $refundId, $refundAmount);
-                self::scheduleRefundTask($orderId, $refundId, $refundAmount);
-            }
-        } catch (\Throwable $exception) {
-            LoggerFactory::create('refund')->error($exception->getMessage(), $exception->getTrace());
-            // This will create a refund record without payment ID, which the OrderRefundTask will use to retry.
-            // This basically means that the newWebPayment api call failed.
-            if (false === $payOrderRefundId) {
-                self::scheduleRefundTask($orderId, $refundId, $refundAmount);
-            }
+        if (!self::refundExists($orderId, $refundId)) {
+            self::addRefund($orderId, null, $refundId, $refundAmount);
         }
-    }
 
-    public static function doCreateRefundPayment($orderId, $refundAmount, $refundId, $storekeeperId)
-    {
-        $api = StoreKeeperApi::getApiByAuthName();
-        $shopModule = $api->getModule('ShopModule');
-        if (self::hasPayment($orderId)) {
-            $storekeeperPaymentId = self::getPaymentId($orderId);
-            $storekeeperRefundId = $shopModule->refundAllOrderItems([
-                'id' => $storekeeperId,
-                'refund_payments' => [
-                    [
-                        'payment_id' => $storekeeperPaymentId,
-                        'amount' => -abs($refundAmount),
-                        'description' => sprintf(
-                            __('Refund via Wordpress plugin (Refund #%s)', I18N::DOMAIN),
-                            $refundId
-                        ),
-                    ],
-                ],
-            ]);
-            $payOrderRefundId = self::addRefund($orderId, $storekeeperRefundId, $refundId, $refundAmount);
-            self::markRefundAsSynced($orderId, $storekeeperRefundId, $refundId);
-        } else {
-            $payOrderRefundId = self::createRefundAsPayment($orderId, $refundId, $refundAmount);
-        }
+        $orderHandler = new OrderHandler();
+        $task = $orderHandler->create($orderId);
 
-        return $payOrderRefundId;
+        if (!$task) {
+            throw new \Exception('Order export task was not created');
+        }
     }
 
-    public static function createRefundAsPayment($orderId, $refundId, $refundAmount)
+    public static function createRefundAsPayment($refundId, $refundAmount)
     {
         $api = StoreKeeperApi::getApiByAuthName();
         $paymentModule = $api->getModule('PaymentModule');
-        $storekeeperRefundId = $paymentModule->newWebPayment([
-            'amount' => -abs($refundAmount), // Refund should be negative
+
+        return $paymentModule->newWebPayment([
+            'amount' => round(-abs($refundAmount)), // Refund should be negative
```

Comment:
```suggestion
            'amount' => round(-abs($refundAmount),2), // Refund should be negative
```
cents are also important

Reply:
Changes implemented.